### PR TITLE
feat: [CDS-86103]: user license view ui changes

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

Updated license inf
Screenshot 2023-12-06 at 6 53 15 AM
o:

### CODE REVIEW

**Review by satyam-harness — DISMISSED** [review]

File: src/modules/27-platform/auth-settings/pages/subscriptions/overview/CDUsageInfo.tsx [hunk 1]
```diff
@@ -68,11 +76,46 @@ const ActiveServiceCard: React.FC<{ subscribedIns: number; activeIns: number; di
...
+            </Text>
+            {getInfoIcon(tooltipName)}
+          </Layout.Horizontal>
+        </Layout.Horizontal>
+        <PercentageBar width={percentUsed} />
+        <Layout.Horizontal flex={{ justifyContent: 'space-between' }}>
+          <Text font={{ size: 'small' }}>
+            {computedCount} {getString('common.subscribed')}
+          </Text>
+          <Text>{Math.ceil(count / workloads)}%</Text>
```
Comment (abhinavrastogi-harness) [inline src/modules/27-platform/auth-settings/pages/subscriptions/overview/CDUsageInfo.tsx:105]: `workloads` defaults to `0`.. which will lead to division by zero error, and display `Infinity%` probably or crash.

File: src/modules/27-platform/auth-settings/pages/subscriptions/overview/CDUsageInfo.tsx [hunk 2]
```diff
@@ -96,22 +139,42 @@ const CDUsageInfo: React.FC<SubscriptionUsageProps> = props => {
...
       <Layout.Horizontal spacing="large">
-        <ActiveServiceCard
-          subscribedIns={licenseDataInfo.workloads || 0}
-          activeIns={usage?.cd?.serviceLicenses?.count || 0}
-        />
+        {hasUserCDLicenseView && workloadsCnt > MAX_LICENSES ? (
+          <ActiveUsageInfoCard
+            workloads={workloadsCnt}
+            count={usage?.cd?.serviceLicenses?.count || 0}
+            computedCount={Math.ceil((usage?.cd?.serviceLicenses?.count || 0) / 3)}
```
Comment (abhinavrastogi-harness) [inline src/modules/27-platform/auth-settings/pages/subscriptions/overview/CDUsageInfo.tsx:169]: looks like the only difference in the above and here is the division by `60` or by `3`. can we move that condition here, and avoid duplicating the entire tree?

File: src/modules/27-platform/auth-settings/pages/subscriptions/overview/SubscriptionUtils.tsx [hunk 3]
```diff
@@ -57,6 +59,7 @@ function getExpiryMsg({
   )
 }
 
+export const MAX_LICENSES = 1000000
```
Comment (abhinavrastogi-harness) [inline src/modules/27-platform/auth-settings/pages/subscriptions/overview/SubscriptionUtils.tsx:62]: this looks wrong. why are we hardcoding max licenses in the UI?

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]
how come there's no tests updated for this change? looks like a change in business logic, which should be unit tested. Especially because this is hard to test with real data as it involves licensing corner cases.

File: src/modules/27-platform/auth-settings/pages/subscriptions/overview/SubscriptionUtils.tsx (hunk 3 above)
Comment (sahithikolichala) [reply src/modules/27-platform/auth-settings/pages/subscriptions/overview/SubscriptionUtils.tsx:62]: This was asked by @AnshuH  to have this default Value and compare

File: src/modules/27-platform/auth-settings/pages/subscriptions/overview/CDUsageInfo.tsx (hunk 1 above)
Comment (sahithikolichala) [reply src/modules/27-platform/auth-settings/pages/subscriptions/overview/CDUsageInfo.tsx:105]: not really, this block is executed only when the workloads is greater than MAX_LICENSES

File: src/modules/27-platform/auth-settings/pages/subscriptions/overview/CDUsageInfo.tsx (hunk 2 above)
Comment (sahithikolichala) [reply src/modules/27-platform/auth-settings/pages/subscriptions/overview/CDUsageInfo.tsx:166]: ya, i can refactor this as follow up

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by cjlee01 — APPROVED** [review]
